### PR TITLE
Add super state machine 1.0

### PR DESCRIPTION
**Pull request by ericdill**

### CODE REVIEW

**Comment by ericdill** [comment]
@beregond I am adding super_state_machine to conda-forge for linux, mac and windows. This will allow users to easily install this with [conda] (http://conda.pydata.org/docs/) (conda install -c conda-forge super_state_machine). The maintenance burden here is light as all testing and releases are built and deployed automatically with CI. Changes to how it is being built can be controlled by changing the recipe which will be located at https://github.com/conda-forge/super_state_machine-feedstock shortly after this PR is finished and merged. If you would like to be added as a maintainer (to make new releases for instance), I can happily add you to the maintainers list. If not, that is totally fine too!

File: recipes/super_state_machine/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - enum34  # [py2k]
+    - six
+
+test:
+  imports:
+    - super_state_machine
+
+about:
+  home: https://github.com/beregond/super_state_machine
+  license: BSD License
```
Comment (jakirkham) [inline recipes/super_state_machine/meta.yaml]: Please specify number of clauses and drop "License".

File: recipes/super_state_machine/nuke_enum34.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--- setup.py.orig	2015-06-25 14:35:18.407380506 -0400
++++ setup.py	2015-06-25 14:35:41.724055571 -0400
+@@ -37,7 +37,6 @@
+                  'super_state_machine'},
+     include_package_data=True,
+     install_requires=[
+-        'enum34',
```
Comment (jakirkham) [inline recipes/super_state_machine/nuke_enum34.patch]: Could we please add and issue/PR upstream about this issue and xref it here?

File: recipes/super_state_machine/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+package:
+  name: super_state_machine
+  version: {{ version }}
+
+source:
+  url: https://github.com/beregond/super_state_machine/archive/{{ version }}.tar.gz
+  fn: suitcase-{{ version }}.tar.gz
+  sha256: a4aac93307f381c64e25b24de347b0527608cc551b66f41c34b2f2df80b445b3
+  patches:
+    - nuke_enum34.patch  # [py3k]
```
Comment (jakirkham) [inline recipes/super_state_machine/meta.yaml]: Also, please cite the issue/PR that this is related to.

File: recipes/super_state_machine/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipes/super_state_machine/meta.yaml]: @ericdill this seems to be BSD 3-Clause BTW. See https://github.com/beregond/super_state_machine/blob/master/LICENSE

File: recipes/super_state_machine/meta.yaml (hunk 3 above)
Comment (ocefpaf) [inline recipes/super_state_machine/meta.yaml]: Do you need this patch? I thought that `--single-version-externally-managed` would prevent anything listed in the `install_requires` to be installed by default.

Comment (ericdill) [inline recipes/super_state_machine/meta.yaml]: @ocefpaf Neat!  Thanks for pointing that out. I definitely did not realize that was the case.  I can get rid of the patch

**Comment by ericdill** [comment]
@jakirkham Thanks for taking the time to review this.  I unfortunately missed your comments go by. Sorry about that!

**Comment by ericdill** [comment]
Welp. That's a new one... appveyor is failing on an encoding problem in setup.py for python 3.4 and 3.5

```
C:\Miniconda\conda-bld\work\super_state_machine-1.0>python setup.py install --single-version-externally-managed --record=record.txt 
  File "setup.py", line 2
SyntaxError: encoding problem: utf8
```

**Comment by beregond** [comment]
@ericdill thanks :)

Unfortunatelly time won't allow me to help you with maintenance. Anyway thanks for this PR :)

**Comment by patricksnape** [comment]
Try [this](https://bugs.python.org/msg233129) for the encoding.

**Comment by ericdill** [comment]
@patricksnape Yeah I just tried to do that via patching and it seems to still be failing on windows.  Oh, duh. I actually need to list the patches under source/patch

**Comment by ericdill** [comment]
Neat, this seems to be working now. Thanks for the input @jakirkham @patricksnape. I'll put in a PR against super state machine to fix the file encoding issue and update the feedstock once that is merged (or not update it if the PR is denied)

**Comment by ocefpaf** [comment]
@ericdill is this good to go?

**Comment by ericdill** [comment]
@ocefpaf yes this is ready. thanks!

**Comment by ericdill** [comment]
Thanks @ocefpaf !

**Pull request merged**